odl/operator/operator.py

```diff
@@ -1328,11 +1332,14 @@ def derivative(self, x):
             Evaluation point of the derivative. Needs to be usable as
             input for the ``right`` operator.
         """
-        left_deriv = self.left.derivative(self.right(x))
-        right_deriv = self.right.derivative(x)
+        if self.is_linear:
+            return self
+        else:
+            left_deriv = self.left.derivative(self.right(x))
```

Comment:
This is where it amkes the largest difference, since we need to call `self.right(x)`, which gives an extra operator evaluation.